src/mpid/ch4/shm/xpmem/globals.c

```diff
@@ -15,12 +15,10 @@ MPL_dbg_class MPIDI_CH4_SHM_XPMEM_GENERAL;
 #endif
 
 /* Preallocated segment objects */
-MPIDI_XPMEM_seg_t MPIDI_XPMEM_seg_mem_direct[MPIDI_XPMEM_SEG_PREALLOC] = { {0}
-};
+MPIDI_XPMEM_seg_t MPIDI_XPMEM_seg_mem_direct[MPIDI_XPMEM_SEG_PREALLOC];
 
 /* Preallocated counter objects */
-MPIDI_XPMEM_cnt_t MPIDI_XPMEM_cnt_mem_direct[MPIDI_XPMEM_CNT_PREALLOC] = { {0}
-};
+MPIDI_XPMEM_cnt_t MPIDI_XPMEM_cnt_mem_direct[MPIDI_XPMEM_CNT_PREALLOC];
 
```

Comment:
Is the automatic-zeroed assumption portable cross different platforms? If not, I'd keep the original code for correctness.

Reply:
Static global variables are always initialized to zero, that is in the C standards. I looked it up once :)

We have one situation that compiler will place the global variables to different section and it will cause issue linking with fortran common symbols. In this case, I believe it is safe.

Reply:
I believe the reason we put this in was because it was causing warnings without it (unless I'm remembering incorrectly). If you compile with `-Wall`, do you see any new warnings here?

Reply:
Here is the strict build log: https://jenkins-pmrs.cels.anl.gov/job/mpich-review-ch4-xpmem/30/compiler=gnu,jenkins_configure=strict,label=xpmem,netmod=ofi,shmmod=xpmem/consoleFull